### PR TITLE
Split launcher config into static and custom

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy [hunk 1]
```diff
@@ -74,7 +74,7 @@ class JavaDistributionPlugin implements Plugin<Project> {
 
         LaunchConfigTask launchConfig = project.tasks.create('createLaunchConfig', LaunchConfigTask, {
             group = GROUP_NAME
-            description = "Generates launcher.yml configuration."
+            description = "Generates launcher-static.yml configuration."
```
Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy]: this really generates both launcher configs, right?

File: src/main/groovy/com/palantir/gradle/javadist/LaunchConfigTask.groovy [hunk 2]
```diff
@@ -33,11 +32,11 @@ class LaunchConfigTask extends AbstractTask {
 
     @TaskAction
     void createConfig() {
-        writeConfig(createConfig(ext.args), "launch/launcher.yml")
+        writeConfig(createConfig(ext.args), "launch/launcher-static.yml")
```
Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/javadist/LaunchConfigTask.groovy]: static launch config should probably go to service/bin and remain immutable

Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/javadist/LaunchConfigTask.groovy]: or maybe it doesn't matter?

File: readme.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ As part of package creation, this plugin will create three shell scripts:
    the defined `mainClass`. This script is considered deprecated due to security issues with
    injectable Bash code; use the javalauncher binaries instead (see below).
  * `service/bin/javalauncher-<architecture>`: native binaries for executing the specified `mainClass`,
-   configurable via `var/launch/launcher.yml`. 
+   configurable via `var/launch/launcher-static.yml` and `var/conf/launcher-custom.yml`.
```
Comment (markelliot) [inline readme.md]: we need to document the format of these files most likely, and define what's overridable in launcher-custom

File: src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy (hunk 1 above)
Comment (uschi2000) [inline src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy]: it generates the check one the static one, but not the custom one.

File: src/main/groovy/com/palantir/gradle/javadist/LaunchConfigTask.groovy (hunk 2 above)
Comment (uschi2000) [inline src/main/groovy/com/palantir/gradle/javadist/LaunchConfigTask.groovy]: happy to go back to `service/bin`

File: readme.md (hunk 3 above)
Comment (uschi2000) [inline readme.md]: documented here: https://github.palantir.build/rfink/go-java-launcher
will link once that's OSS

**Comment by uschi2000** [comment]
done

**Pull request merged**